### DISCUSSION TITLE
Vips::Image.thumbnail and Vips::Image.thumbnail_image resize to height instead of width when no hash options are present

### DISCUSSION THREAD

**Discussion by i-h8-github** [libvips/libvips#4312] [Q&A]
## Bug report

**Describe the bug**
`Vips::Image.thumbnail` and `Vips::Image.thumbnail_image` resize to height instead of width when no hash options are present. If a `:height` is also supplied, they will resize to the expected width

**To Reproduce**
```ruby
img = Vips::Image.new_from_file "file.pdf"
puts "#{img.width}, #{img.height}" # 612, 859
thumb = img.thumbnail_image(200)
puts "#{thumb.width}, #{thumb.height}" # 142, 200
thumb = img.thumbnail_image(200, :height => 1000)
puts "#{thumb.width}, #{thumb.height}" # 200, 281

thumb = Vips::Image.thumbnail "file.pdf", 200
puts "#{thumb.width}, #{thumb.height}" # 143, 200 — Different height from above for some reason
thumb = Vips::Image.thumbnail "file.pdf", 200, :height => 1000
puts "#{thumb.width}, #{thumb.height}" # 200, 281
```
**Expected behavior**
Should resize to specified width as described

**Actual behavior**
Resizes to a height of the specified width

**Environment**
 - OS: Ubuntu 24.04
 - Vips: 8.16.0

**Comment by jcupitt** [comment]
Cross-posted here: https://github.com/libvips/ruby-vips/issues/408

With one parameter, `thumbnail` resizes to fit within a width x width box, docs are here:

https://www.libvips.org/API/current/libvips-resample.html#vips-thumbnail

**Comment by i-h8-github** [comment]
Thank you for your time, attention, and explanation.

That is not how ruby-vips is documented, though.

https://libvips.github.io/ruby-vips/Vips/Image.html#thumbnail_image-instance_method

**Reply by i-h8-github** [reply]
Is there an explanation for why `thumbnail` and `thumbnail_image` produce images of different heights?

**Reply by jcupitt** [reply]
I don't think they do, do they? Do you have an example where they differ?

(please don't use `thumbnail_image`!)

**Reply by i-h8-github** [reply]
> I don't think they do, do they? Do you have an example where they differ?
> 
> (please don't use `thumbnail_image`!)

Do you mean not to use `thumbnail_image` as an example? Or not to use it at all?

In my example above at line 9, they produce different heights.
